docs/user_guide/fairness_in_machine_learning.rst

```diff
@@ -141,6 +141,199 @@ which groups these values were observed, as well as the difference and ratio
+this realm. This trap also occurs as definitions of fairness are
+contested and changed and models become obsolete.
+
+Example: consider the problem of internet connectivity in rural communities.
+An example of the solutionism trap is assuming that by using data science to
+study internet speed in a given region, policies and deals can be arranged to
+reduce the price of internet access. However, if there are additional socioeconomic
+problems within a community, for example lack of education, infrastructure, information
+technology and health services, then an algorithmic solution purely focused on internet
+speed will fail to meaningfully address the needs of the community.
+
+As an example, the Federation of Canadian municipalities released a report
+ tackling both connectivity and infrastructure needs of rural communities to
+more meaningfully address this social issue
+https://fcm.ca/sites/default/files/documents/resources/report/broadband-access-rural-canada.pdf.
+
+The Ripple Effect Trap
+^^^^^^^^^^^^^^^^^^^^^^
+
+This trap occurs when we do not study the unintended consequences of introducing
+technology into an existing social system. Such consequences include change in
+behaviour, outcomes individual experience and a change in underlying social values
+and incentives of a given social system, for instance by increasing perceived value
+of quantifiable metrics over non-quantifiable ones.
+
+Example: consider the problem of banks deciding whether an individual should
+be approved for a loan. Prior to using machine learning algorithms
+to compute a "score", banks might rely on engaging in conversations with
+clients, recommending a plan based on their unique situation, and
+discussing with other team members to obtain feedback. By introducing an
+algorithm, it is possible that banks stop engaging in conversations
+with team members and clients, and assume the algorithm is good enough
+to blindly accept a recommendation. Furthermore, some might use
+information on people algorithm typically rejects to create a system
+that takes advantage of people who need funds but did not receive them.
+
+To avoid this pitfall, we must be aware that once a technology is incorporated
+into a social context, new groups may reinterpret it and potentially use this
+information for outside purposes which include nefarious ones. To
+solve for this issue, we must incorporate "what if" scenarios into testing
+of the model.
+
+The Formalism Trap
+^^^^^^^^^^^^^^^^^^
+
+This trap occurs when abstraction results in failure to take
+into account the full picture behind fairness in a situation. Many tasks of a data
+scientist involve some form of formalization: from measuring real-world phenomena as
+data to translating business KPI's and constraints into metrics, loss functions, or parameters.
+We fall into the formalism trap when this translation fails to account the social dimensions
```

Comment:
```suggestion
We fall into the formalism trap when we fail to account for the full meaning of social concepts like fairness. This occurs because there is no purely mathematical way to resolve conflicting definitions of fairness. This is also because fairness is complex and contested by social actors, and it cannot only be captured mathematically, but needs to be understood procedurally (i.e., [insert explanation here]) and situated in social contexts.
```

Reply:
I think in a previous review I wrote a suggested version for this explanation (I'll see if I can find it and tag you in it). If it's easier, feel free to just use that language directly!